src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt

```diff
@@ -172,7 +172,7 @@ private constructor(
       if (proxy.PROXY_TYPE_IS_SOCKS) {
         processBuilder.environment()["HTTP_PROXY"] = "socks://$proxyUrl"
       }
-      if (proxy.USE_HTTP_PROXY) {
+      else {
```

Comment:
I do not understand this change, old code looks valid to me and proposed change seems to be incorrect.
Could you please elaborate on the proposed logic change?